ServerCore/Pages/Submissions/Index.cshtml

```diff
@@ -457,7 +457,18 @@ else
             }
         </div>
         <div class="inlinewrapper">
-            @if (!Model.PuzzleState.IsEmailOnlyMode)
+            @if (!Model.Event.HideHints)
+            {
+                if (Model.Puzzle.IsForSinglePlayer)
+                {
+                    <div><b>Need help?</b> <a asp-page="/Puzzles/SinglePlayerPuzzleHints" asp-route-puzzleid="Model.Puzzle.ID">Click this link</a> to access hints for thsi puzzle.</div>
```

Comment:
Typo:  thsi -> this in both links

Also, I'd prefer `target="_blank"` be added to both these links as well so they open in new tabs and don't potentially erase progress.